com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingMain.raytrace

```diff
@@ -90,68 +89,90 @@ void MissMaterial(inout PathIntersection pathIntersection : SV_RayPayload)
-    float4 jitteredPixelCoord = float4(currentPixelCoord, 1.0, 1.0);
-    jitteredPixelCoord.x += GetSample(currentPixelCoord, _RaytracingSampleIndex, 40);
-    jitteredPixelCoord.y += GetSample(currentPixelCoord, _RaytracingSampleIndex, 41);
+    // Get the current tile coordinates (for interleaved tiling)
+    uint2 tileCount = uint2(_PathTracingTilingParameters.xy);
+    uint2 tileIndex = uint2(_PathTracingTilingParameters.zw);
 
-    // Compute the ray direction from those coordinates (for zero aperture
-    float3 directionWS = -normalize(mul(jitteredPixelCoord, _PixelCoordToViewDirWS).xyz);
-    float3 cameraPosWS = GetPrimaryCameraPosition();
+    // Jitter them (we use 4x10 dimensions of our sequence during path tracing atm, so pick the next available ones)
+    float4 jitteredPixelCoord = float4(pixelCoord, 1.0, 1.0);
+    jitteredPixelCoord.x += GetSample(pixelCoord, _RaytracingSampleIndex, 40) / tileCount.x;
+    jitteredPixelCoord.y += GetSample(pixelCoord, _RaytracingSampleIndex, 41) / tileCount.y;
 
-    float apertureRadius = _PathTracedDoFConstants.x;
-    if (apertureRadius > 0.0)
-    {
-        // Compute the ray origin and direction for a lens with non-zero aperture
+    // Create the ray descriptor for this pixel
+    RayDesc ray;
+    ray.TMin = _RaytracingCameraNearPlane;
+    ray.TMax = FLT_INF;
 
-        // Sample the lens apperture using the next available dimensions (we use 40 for path tracing, 2 for sub-pixel jittering, 64 for SSS -> 106, 107)
-        float r1 = GetSample(currentPixelCoord, _RaytracingSampleIndex, 106);
-        float r2 = GetSample(currentPixelCoord, _RaytracingSampleIndex, 107);
-        float2 uv = apertureRadius * SampleDiskUniform(r1, r2);
+    // We need the camera forward direction in both types of projection
+    float3 cameraDirection = GetViewForwardDir();
 
-        // Compute the new ray origin ( _ViewMatrix[0] = right, _ViewMatrix[1] = up, _ViewMatrix[2] = forward )
-        cameraPosWS += _ViewMatrix[0].xyz * uv.x + _ViewMatrix[1].xyz * uv.y;
+    // Compute the ray's origin and direction, for either perspective or orthographic projection
+    if (IsPerspectiveProjection())
+    {
+        ray.Origin = GetPrimaryCameraPosition();
+        ray.Direction = -normalize(mul(jitteredPixelCoord, _PixelCoordToViewDirWS).xyz);
 
-        // Compute the focus point by intersecting the pinhole ray with the focus plane
-        float focusDistance = _PathTracedDoFConstants.y;
-        float t = focusDistance / dot(directionWS, _ViewMatrix[2].xyz);
-        float3 focusPointWS = GetPrimaryCameraPosition() - t * directionWS;
+        // Use planar clipping, to match rasterization
+        float dotDirection = dot(cameraDirection, ray.Direction);
+        ray.TMin /= dotDirection;
 
-        // The new ray direction should pass through the focus point
-        directionWS = normalize(focusPointWS - cameraPosWS);
+        ApplyDepthOfField(pixelCoord, dotDirection, ray.Origin, ray.Direction);
```

Comment:
Should we use here "pixelCoord * tileCount + tileIndex" instead of "pixelcoord" to avoid correlation in lens sampling? Or perhaps I missed something?

Reply:
No you're right, thanks! I'll fix that.